roles/ceph-osd/tasks/bcache.yml

```diff
@@ -1,4 +1,35 @@
 ---
+# Create hybrid bucket
+- name: Create root hybrid-bucket root
+  shell: ceph osd crush add-bucket {{ hybrid_pool }} root
+  run_once: true
+
+# Create crush ruleset for the hybrid root bucket
+- name: Create crush ruleset hybrid root bucket
+  shell: ceph osd crush rule create-simple {{ hybrid_ruleset }} {{ hybrid_pool }} host firstn
+  run_once: true
+
+# Create osd host bucket
+- name: Create osd host bucket
+  shell: ceph osd crush add-bucket {{ ansible_hostname }} host
+
+# Move the host bucket under corresponding root
+- name: Move host bucket under root bucket
+  shell: ceph osd crush move {{ ansible_hostname }} root={{ hybrid_pool }}
+
```

Comment:
this command can be executed multi times. and return code is always 0. I think this is what ceph community wants, they want to make their command idempotent.